api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -803,20 +799,13 @@ public DbUser syncTwoFactorAuthStatus(DbUser targetUser, Agent agent, boolean is
   }
 
   @Override
-  public DbUser syncDuccVersionStatus(
-      DbUser targetUser, Agent agent, @Nullable Integer signedDuccVersion) {
+  public DbUser syncDuccVersionStatus(DbUser targetUser, Agent agent) {
     if (isServiceAccount(targetUser)) {
       // Skip sync for service account user rows.
       return targetUser;
     }
 
-    // convert Integer to int to prevent a NPE from comparison-unboxing
-    final int signedVersionForComparison =
-        Optional.ofNullable(signedDuccVersion)
-            // null is invalid, so convert to a known-invalid int
-            .orElse(accessModuleService.getCurrentDuccVersion() - 1);
-
-    if (signedVersionForComparison != accessModuleService.getCurrentDuccVersion()) {
+    if (!accessModuleService.hasUserSignedTheCurrentDucc(targetUser)) {
       accessModuleService.updateCompletionTime(
```

Comment:
If the user has already signed ducc, do we need to update completion time to current time so that the expiry date time can be extended as well?

Reply:
This is an offline sync task, not driven by user action.  It would not be correct to say that the user signed the DUCC when this runs.